antenna/crashmover.py

```diff
@@ -2,54 +2,43 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at https://mozilla.org/MPL/2.0/.
 
-from collections import deque
+from dataclasses import dataclass
 import logging
-import time
 
 from everett.manager import Option, parse_class
-from gevent.pool import Pool
 import markus
 
-from antenna.heartbeat import register_for_life, register_for_heartbeat
-from antenna.util import isoformat_to_time
+from antenna.util import MaxAttemptsError, retry
 
 
 LOGGER = logging.getLogger(__name__)
 MYMETRICS = markus.get_metrics("crashmover")
 
+#: Maximum number of attempts to save or publish a crash before we give up
+MAX_ATTEMPTS = 5
 
-#: Maximum number of attempts to save a crash before we give up
-MAX_ATTEMPTS = 20
+#: Seconds to sleep between attempts to save or publish a crash
+RETRY_SLEEP_SECONDS = 2
 
-#: SAVE and PUBLISH states of the crash mover
-STATE_SAVE = "save"
-STATE_PUBLISH = "publish"
 
+def _incr_wait_generator(
+    counter, attempts=MAX_ATTEMPTS, sleep_seconds=RETRY_SLEEP_SECONDS
+):
+    def _generator_generator():
+        for _ in range(attempts - 1):
+            MYMETRICS.incr(counter)
+            yield sleep_seconds
```

Comment:
`MYMETRICS.incr(counter)` gets called when `antenna.util.retry` peeks a value (but only once per value, not once per peek). This is means that technically if `antenna.util.retry` peeks on the value without using it, then we would over-count, but right now it doesn't behave that way. An alternative would be specifying sleep_function and emitting this metric there.